hail/python/hailtop/batch/backend.py

```diff
@@ -759,7 +767,16 @@ async def compile_job(job):
             print(f'Built DAG with {n_jobs_submitted} jobs in {round(time.time() - build_dag_start, 3)} seconds.')
 
         submit_batch_start = time.time()
-        batch_handle = bc_batch.submit(disable_progress_bar=disable_progress_bar)
+        if batch._batch_handle is None:
+            batch_handle = bc_batch_builder.submit(disable_progress_bar=disable_progress_bar)
+            batch._batch_handle = batch_handle
+        else:
+            new_batch_handle = bc_batch_builder.submit(disable_progress_bar=disable_progress_bar)
+            assert batch._batch_handle.id == new_batch_handle.id
```

Comment:
I think if we make the above change where we call `update_batch` with the batch_handle, then we can actually make the following assertion:

```python
batch._batch_handle == new_batch_handle
```

which seems necessary since the aioclient Batch and Job classes hold references to each other and we don't want to run into situations where the aioclient Batch object doesn't have a reference to all past jobs.